src/data_store/data_store_conduit.cpp

```diff
@@ -55,6 +60,7 @@ data_store_conduit::data_store_conduit(
   m_world_master = m_comm->am_world_master();
   m_trainer_master = m_comm->am_trainer_master();
   m_rank_in_trainer = m_comm->get_rank_in_trainer();
+  m_rank_in_world = m_comm->get_rank_in_world();
   m_np_in_trainer = m_comm->get_procs_per_trainer();
```

Comment:
Why are we duplicating information?

(This should probably be addressed in a separate PR.)

Reply:
will adress in a separate PR: too many places to make changes, and not relevant to this PR's purpose.